src/mdbExtensionController.ts

```diff
@@ -289,21 +299,24 @@ export default class MDBExtensionController implements vscode.Disposable {
           vscode.window.showErrorMessage(
             'Please connect to this connection before adding a database.'
           );
-          return false;
+
+          return Promise.resolve(false);
         }
 
         if (this._connectionController.isDisconnecting()) {
           vscode.window.showErrorMessage(
             'Unable to add database: currently disconnecting.'
           );
-          return false;
+
+          return Promise.resolve(false);
```

Comment:
Why do we wrap the false with the `Promise.resolve(` here and in a few other places?